prettier/src/utils/tokenize.ts

```diff
@@ -0,0 +1,160 @@
+const tokenPlaceholderMap = new Map();
+let tokenIndex = 0;
+
+const getTokenByValue = (match, type) => {
+  let tokens = tokenPlaceholderMap.entries();
+  for (let token of tokens) {
+    if (token[1].type === type && token[1].token === match) {
+      return token[0];
+    }
+  }
+};
+
+export const tokenize = (input) => {
```

Comment:
As I've mentioned before, this step replaces all of the HubL variables, tags, and comments with placeholders. Placeholders vary depending on the type and location of a string. During this refactor I've tried to make the types and placeholder strings more obvious.